app/src/main/java/com/pvp/app/ui/screen/friends/FriendsViewModel.kt

```diff
@@ -0,0 +1,161 @@
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.pvp.app.api.FriendService
+import com.pvp.app.api.UserService
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class FriendsViewModel @Inject constructor(
+    private val friendService: FriendService,
+    private val userService: UserService
+) : ViewModel() {
+
+    val toastMessage = mutableStateOf<String?>(null)
+
+    private val _isRequestSent = MutableStateFlow(false)
+    val isRequestSent = _isRequestSent.asStateFlow()
+
+    val user = userService.user.stateIn(
+        viewModelScope,
+        started = SharingStarted.WhileSubscribed(5000),
+        initialValue = null
+    )
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    val friendObject = user.flatMapLatest { user ->
+        user?.email?.let { email ->
+            friendService.create(email)
+            friendService.get(email)
+        } ?: flowOf(null)
+    }.stateIn(
+        viewModelScope,
+        started = SharingStarted.WhileSubscribed(5000),
+        initialValue = null
+    )
+
+    fun create(email: String) {
+        viewModelScope.launch {
+            friendService.create(email)
+        }
+    }
+
+    fun addFriend(friendEmail: String) {
+        viewModelScope.launch {
+            _isRequestSent.value = false
+
+            if (friendEmail.isEmpty()) {
+                toastMessage.value = "Please enter an email"
+                _isRequestSent.value = false
+
+                return@launch
+            }
+
+            val email = user.value?.email ?: return@launch
+
+            val friendObject = friendService
+                .get(email)
+                .firstOrNull()
+                ?: return@launch
+
+            val friendUser = userService
+                .get(friendEmail)
+                .firstOrNull()
+
+            if (friendUser == null) {
+                toastMessage.value = "User with email $friendEmail does not exist"
+                _isRequestSent.value = false
+
+                return@launch
+            }
+
+            friendService.create(friendEmail)
+
+            val toastMessageValue = friendService.addFriend(
+                friendObject,
+                email,
+                friendEmail
+            )
+
+            toastMessage.value = toastMessageValue
+            _isRequestSent.value = (
+                    toastMessageValue == "Friend request sent!" ||
+                    toastMessageValue == "Friend request accepted!"
```

Comment:
If ViewModel's code is hard dependent on what string is returned from the unknown Service implementation, then it should be somehow controlled a little more, so that the changes in the Service implementations would also take place in ViewModel's implementation.

This looks very bad practice, so if you don't want to change this with code, then at least put comments onto the Service interface appropriate methods indicating that returned string is strictly associated with how ViewModel works. So if any changes to the default implementation are made, changes in the ViewModel should also be made.